### DISCUSSION TITLE
Is it necessary to filter gene or cell filter needed after DecontX

### DISCUSSION THREAD

**Discussion by shaoyu1230** [campbio/celda#393] [Q&A]
In Seurat vignettes, we often filter gene and features in CreateSeuratObject.
```r
pbmc <- CreateSeuratObject(counts = pbmc.data, project = "pbmc3k", min.cells = 3, min.features = 200)
``` 
Do we still need to do the filter after running DecontX?

**Comment by joshua-d-campbell** [comment]
Hi @MISAKA-DaYu, thanks for trying out our tool. Yes, should still do all of the same filtering you would normally do. DecontX returns a raw count matrix where the ambient RNA has been subtracted out. All of the other filtering and analysis methods can be applied to this matrix as normal. I am going to move this to a Discussion for future reference.

**Reply by batalha23** [reply]
Hi Joshua, just adding a follow-up question. Then would the correct order when working with the Seurat pipeline be...
1. Run decontX on the dataset (from filtered barcode matrices), without any gene/filtering step
2. Perform regular QC to remove low quality / empty droplets or doublets (like filtering for UMI or gene count, mtDNA, etc)
3. Move forward with normalization, DimRed, etc?

While I do understand the rationale behind substracting ambient RNA as a first step, this doesn't seem to agree with the suggestion of using dataset-informed clustering, which we can only obtain after performing QC based on cell and gene filtering. 

Also, I've noticed I'm only able to run decontX when the number of genes between my dataset/query and the raw/background matrix is the same, which prevents the use of the `min.cells = 3` argument (removing genes expressed in fewer than 3 cells) when initializing my Seurat object. Since the Seurat development team strongly advises _against_ removing genes after this step, I was wondering if you or anyone else has used any workaround for this? The first thing I could think of would be to extract the decontaminated data from the SCE object as a counts matrix and use it as input to CreateSeuratObject(), but since I'm not used to working with SCE objects I'm not sure this could work.

Thanks so much in advance, and thank you for the tool!
